feat: Added new resourceclient to ror client.

402-matiaskj opened the pull request:
The PR does mainly two things: implement Delete for v2 resources and adding a bit of abstraction to the ror client through the resource client. The new client aims to allow convenience methods to be built on top of the basic CRUD operaions provided by the transports.

# ROR API: 

## Resource controller
- added inn some validation from similar controller.
- It was using the old ACL framework
- changed ACL scope to read, was update. The user is doing a read
## Resource controller Delete
- changed mismatched guid error to 400, was 501, This is an error on the users part and should be reflected in the answer.
- return resource update result, was returning string. Lets us better inform the client what was deleted
## Routes
- Deprecate update by uid and get by uid, get and update can already do this, so we can let the client take care of convenience methods like this.

# ROR client

## ROR client
* rorclient no longer has a reference to the resource transport, instead it exposes the new resource client.
## Resource client
![1729798692-24 10-shareX-screen](https://github.com/user-attachments/assets/46de816e-6b57-4f31-9566-dc396bdf7016)
* abstraction layers for the resource transport, implements CRUD and a few convenience methods built on top of the basic operations. The deprecated api routes get by uid and update by uid (now called update one) is implemented here.
## Resttransport http client
* Add delete method
## Resttransport resources
* removed update by uid, get by uid
* implemented delete
* exists still needs to be implemented
## v2 Resources interface
* removed update by uid and get by uid from the interface, transports should only implement basic CRUD
* renamed delete by uid to delete and exists by uid to exists. 
## Resource set
* refactored to keep struct and methods together. 
 Added new resource client

havardelnan commented:
LGTM @rogerwesterbo brekker dette noe i web?

rogerwesterbo commented on pkg/clients/rorclient/transports/resttransport/v2/resources/resources.go:
```diff
@@ -44,24 +45,35 @@ func (c *V2Client) Update(res *rorresources.ResourceSet) (*rorresources.Resource
-	// implementation goes here
-	return false, nil
+func (c *V2Client) Exists(uid string) (bool, error) {
+	//url, err := url.JoinPath(c.basePath, "uid", uid)
```
Still not implemented?

402-matiaskj replied:
Il get to it today, i just wanted to get the PR up

rogerwesterbo commented:
> LGTM @rogerwesterbo brekker dette noe i web?

I dont think so, but the ui components that uses the endpoint is in another branch

402-matiaskj commented:
# ROR api
## Resource controller
- moved the uid check for exists to before the uid is actually used.
- Added in a nil check to deal with a panic that would happen if resource did not exist
- removed exists v2 checking for v1 resources
# ROR Client
## http transport client
- added head request. Head returns both headers and status requests and ingores any body.
## Resource client
- Added get own hashes function
- implemented exists
# K8s agent v2
## ROR k8s Agent v2
* fixed problems related to changes in the ROR client.
